pwndbg/heap/ptmalloc.py

```diff
@@ -53,38 +52,70 @@ def mp(self):
 
         return self._mp
 
+
     @property
     def malloc_chunk(self):
         if not self._malloc_chunk:
             self._malloc_chunk = pwndbg.typeinfo.load('struct malloc_chunk')
         return self._malloc_chunk
 
+
     @property
     def malloc_state(self):
         if not self._malloc_state:
             self._malloc_state = pwndbg.typeinfo.load('struct malloc_state')
         return self._malloc_state
 
+
     @property
     def mallinfo(self):
         if not self._mallinfo:
             self._mallinfo = pwndbg.typeinfo.load('struct mallinfo')
         return self._mallinfo
 
+
     @property
     def malloc_par(self):
         if not self._malloc_par:
             self._malloc_par = pwndbg.typeinfo.load('struct malloc_par')
         return self._malloc_par
 
+    def _spaces_table(self):
+        spaces_table =  [ pwndbg.arch.ptrsize * 2 ]      * 64 \
+                      + [ pow(2, 6) ]                    * 32 \
+                      + [ pow(2, 9) ]                    * 16 \
+                      + [ pow(2, 12) ]                   * 8  \
+                      + [ pow(2, 15) ]                   * 4  \
+                      + [ pow(2, 18) ]                   * 2  \
+                      + [ pow(2, 21) ]                   * 1
+
+        # There is no index 0
+        spaces_table = [ None ] + spaces_table
+
+        # Fix up the slop in bin spacing ( part of libc - they made
```

Comment:
Hmm... do you have some reference for this? Like url, line of particular file or commit hash?

Mayyyybe we can add such info to this comment. However if it is an url it might be broken later...